src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -237,48 +266,25 @@ void ClusterTaskManager::DispatchScheduledTasksToWorkers(
   }
 }
 
-bool ClusterTaskManager::AttemptDispatchWork(const Work &work,
-                                             std::shared_ptr<WorkerInterface> &worker,
-                                             bool *worker_leased) {
-  const auto &task = std::get<0>(work);
-  const auto &spec = task.GetTaskSpecification();
-  RAY_LOG(DEBUG) << "Attempting to dispatch task " << spec.TaskId();
-
-  std::shared_ptr<TaskResourceInstances> allocated_instances(new TaskResourceInstances());
-  bool schedulable = cluster_resource_scheduler_->AllocateLocalTaskResources(
-      spec.GetRequiredResources().GetResourceMap(), allocated_instances);
-  bool dispatched = false;
-  if (!schedulable) {
-    *worker_leased = false;
-    // Spill at most one task from this queue, then move on to the next
-    // queue.
-    int64_t _unused;
-    bool is_infeasible;
-    auto placement_resources = spec.GetRequiredPlacementResources().GetResourceMap();
-    std::string node_id_string = cluster_resource_scheduler_->GetBestSchedulableNode(
-        placement_resources, spec.IsActorCreationTask(), &_unused, &is_infeasible);
-    RAY_CHECK(!is_infeasible)
-        << "Task cannot be infeasible when it is about to be dispatched";
-    if (node_id_string != self_node_id_.Binary() && !node_id_string.empty()) {
-      NodeID node_id = NodeID::FromBinary(node_id_string);
-      Spillback(node_id, work);
-      dispatched = true;
-    }
-  } else {
-    worker->SetBundleId(spec.PlacementGroupBundleId());
-    worker->SetOwnerAddress(spec.CallerAddress());
-    if (spec.IsActorCreationTask()) {
-      // The actor belongs to this worker now.
-      worker->SetLifetimeAllocatedInstances(allocated_instances);
-    } else {
-      worker->SetAllocatedInstances(allocated_instances);
-    }
-    worker->AssignTaskId(spec.TaskId());
-    worker->SetAssignedTask(task);
-    *worker_leased = true;
-    dispatched = true;
+bool ClusterTaskManager::TrySpillback(const Work &work) {
+  const auto &spec = std::get<0>(work).GetTaskSpecification();
+  int64_t _unused;
+  bool is_infeasible;
+  auto placement_resources = spec.GetRequiredPlacementResources().GetResourceMap();
+  std::string node_id_string = cluster_resource_scheduler_->GetBestSchedulableNode(
+      placement_resources, spec.IsActorCreationTask(), &_unused, &is_infeasible);
+
+  // All tasks on the dispatch queue must at least be feasible on the local node.
+  RAY_CHECK(!is_infeasible)
```

Comment:
I know this was there before, but is this actually a safe check? It seems like this could be true if the resources on this node got modified, because of placement groups for example. It would probably be safer to move it to the infeasible queue.

Reply:
Hmm yeah our assumption was the value is static. But it is a good point about that placement group can change the story here. Potentially related issue; https://github.com/ray-project/ray/issues/14092

But do you think it makes sense to handle it in a different PR? Removing check from here probably could cause other issues.

Reply:
Yes, let's fix it in a separate PR. I'll make a note on the issue about the potential cause.

Reply:
I suspect that will have an easy fix, but I agree it looks problematic in the placement group case, gut we shouldn't solve it in this PR (I'll add a TODO though)